src/mbed_tools/project/_internal/project_data.py

```diff
@@ -97,12 +98,24 @@ def from_new(cls, root_path: Path) -> "MbedProgramFiles":
         )
 
     @classmethod
-    def from_existing(cls, root_path: Path) -> "MbedProgramFiles":
+    def from_existing(
+        cls,
+        root_path: Path,
+        target_name: Optional[str] = None,
+        profile: str = "develop",
+        toolchain: Optional[str] = None,
+    ) -> "MbedProgramFiles":
         """Create MbedProgramFiles from a directory containing an existing program.
 
         Args:
             root_path: The path containing the MbedProgramFiles.
+            target_name: Name of the Mbed target in use.
+            profile: Current profile for build.
+            toolchain: Name of the toolchain in use.
         """
+        if target_name is None or toolchain is None:
+            default_cmake_config = root_path / CMAKE_CONFIG_FILE_PATH
+            target_name, toolchain = _prev_target_toolchain(default_cmake_config)
```

Comment:
I think we should just make users pass the target and toolchain on the command line instead of reading values from mbed_config.cmake.